libkbfs/folder_block_ops.go

```diff
@@ -290,18 +279,19 @@ func (fbo *folderBlockOps) getBlockHelperLocked(ctx context.Context,
 		return nil, InvalidBlockRefError{ptr.Ref()}
 	}
 
-	if block, err := fbo.getBlockFromDirtyOrCleanCache(
-		ptr, branch); err == nil {
+	if block, err := fbo.config.DirtyBlockCache().Get(
+		fbo.id(), ptr, branch); err == nil {
+		return block, nil
+	}
+	if block, err := fbo.config.BlockCache().Get(ptr); err == nil {
+		fbo.config.BlockOps().Prefetcher().PrefetchAfterBlockRetrieved(block, kmd, defaultOnDemandRequestPriority)
```

Comment:
The difference is the priority. When we go through this flow, even if the block was cached in the past, we need to handle it as if it's an on-demand request so that its downstream prefetches are triggered correctly. I'll add a comment to that effect.